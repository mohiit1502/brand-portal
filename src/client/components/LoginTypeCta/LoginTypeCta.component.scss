@import "./../../styles/override/variables";

.c-LoginTypeCta {
  color: #444;
  .page-container {
    .logoBox {
      width: 30%;
      height: 100%;
      min-width: 315px;
      z-index: 99;
      display: block;
      text-align: center;
      background-color: #041e42;
      position: fixed;
      top: 0;
      left: 0;
      background-size: 180%;
      background-position: 0;
      background-repeat: no-repeat;
      .logo {
        margin-top: 100px;
        height: 42px;
        //margin-left: calc(((70% - 480px) / 2) + 30%);
      }
    }

    .main-container {
      position: absolute;
      margin: 100px auto auto calc(((70% - 480px) / 2) + 30%);

      .content-container {
        border: 1px solid #e7e7e7;
        box-shadow: 0 0 5px 1px #e7e7e7;
        background-color: #fff;

        .tab-content {
          max-width: 25rem;
          .error-message {
            margin-bottom: unset;
          }
          .close-button {
            position: absolute;
            font-size: 2rem;
            font-weight: 700;
            text-shadow: 0 1px 0 #fff;
            cursor: pointer;
            color: #888;
            top: 0.8rem;
            right: 1rem;
            line-height: 1.5rem;
          }
        }

        .page-header {
          width: 100%;
          display: block;
          font-weight: bold;
          font-size: 1.75rem;
        }
        .sub-title {
          font-size: 14px;
        }
        .app-btn {
          width: 100%;
          background: $falcon-blue;
          border-radius: 2px;
          padding: 0.75rem 1rem;
          border: none;
          color: white;
          font-size: 14px;
          transition: all 0.3s;

          &.secondary-btn {
            background: white;
            color: $falcon-blue;
            border: 1px solid $falcon-blue;

            &:hover {
              background: $falcon-blue;
              color: white;
            }
          }

          &:hover {
            text-decoration: none;
            background-color: #1292ed;
          }
        }
        .OrSeparator {
          hr {
            display: block;
            height: 1px;
            border: 0;
            border-top: 1px solid #ccc;
            padding: 0;

            &:nth-child(1), &:nth-child(3) {
              flex: 1;
              margin: auto;
            }
            &:nth-child(1) {
              margin-right: 10px;
            }
            &:nth-child(3) {
              margin-left: 10px;
            }
          }
        }
      }
    }

    .footnote-container {
      font-size: 0.75rem;
      .policy-link {
        text-decoration: underline;
        color: #444;
        &:hover {
          color: #1292ed;;
        }
      }
    }
  }
}
