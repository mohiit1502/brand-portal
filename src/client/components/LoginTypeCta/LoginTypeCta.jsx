/* eslint-disable no-shadow, filenames/match-regex */
import React, {useEffect, useState} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {useLocation} from "react-router-dom";
import {dispatchLoginAction, dispatchRegisterAction} from "../../actions";
import {BPDarkLogo} from "./../../images";
import "./LoginTypeCta.component.scss";
import CONSTANTS from "../../constants/constants";

/* eslint-disable complexity */
const LoginTypeCta = props => {
  const {action, dispatchLoginAction, dispatchRegisterAction} = props;
  const location = useLocation();
  const [error, setError] = useState();

  useEffect(() => {
    if (!action && location.pathname) {
      if (location.pathname.endsWith("/login")) {
        dispatchLoginAction();
        if (location.search) {
          const index = location.search.lastIndexOf("=");
          const error = index > -1 ? location.search.substring(index + 1) : location.search;
          setError(error === "true");
        }
      } else if (location.pathname.endsWith("/register")) {
        dispatchRegisterAction();
      }
    }
  }, [action, location]);

  /* eslint-disable no-nested-ternary */
  return (<div className="c-LoginTypeCta">
      <div className="page-container css-0 d-flex flex-column">
        <div className="logoBox">
          <img src={BPDarkLogo} alt="WBP Dark Logo" className="logo" />
        </div>
        <div className={`main-container position-absolute${!action && " loader"}`}>
          <div className="content-container">
            <div className="padder p-4">
              <div className="tab-content text-center">
                {error && <div className="cta-error">
                  <p className="error-message">Your Walmart Brand Portal account has been linked to your Seller Center username and password. To access Brand Portal, use your Seller Center log in.</p>
                  <span className="close-button" onClick={() => setError(false)}>Ã—</span>
                </div>}
                <span className="page-header mb-2"><span>{action ? action === "login" ? "Log In" : "Register" : ""}</span></span>
                <p className="sub-title mb-0">Streamline {action === "login" ? "login" : "registration"} by using one of your existing</p>
                <p className="sub-title">Walmart accounts.</p>
                <a href={`${CONSTANTS.URL.LOGIN_REDIRECT}?clientType=seller`} className="app-btn d-block mt-2">{action ? action === "login" ? "Use Seller Center Account" : "Use Seller Center Account" : ""}</a>
                <div className="OrSeperator w-100 d-flex my-4">
                  <hr /><span>OR</span><hr />
                </div>
                {action === "register" && <p className="sub-title">Don't have a Walmart account?</p>}
                <a href={action === "login" ? `${CONSTANTS.URL.LOGIN_REDIRECT}?clientType=supplier` : `${CONSTANTS.URL.REGISTER_REDIRECT}?clientType=supplier`} className="app-btn d-block secondary-btn mt-2">{action ? action === "login" ? "Use Brand Portal log in" : "Register with Brand Portal" : ""}</a>
              </div>
            </div>
          </div>
          <div>
            <div className="footnote-container mt-4">
              <span className="footnote d-block text-center">
                <span>We recommend setting your monitor resolution to at least 1024 x 768</span>
              </span>
              <span className="footnote d-block text-center">
                <span>U.S. Users: Please use Chrome v49 or newer</span>
              </span>
              <span className="footnote d-block text-center"><span>International Users: Please use Internet Explorer</span>
              </span>
              <span className="footnote d-block text-center">
                <a href="https://corporate.walmart.com/privacy-security" className="policy-link" target="_blank">Privacy Policy</a>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>);
};

LoginTypeCta.propTypes = {
  action: PropTypes.string,
  dispatchLoginAction: PropTypes.func,
  dispatchRegisterAction: PropTypes.func
};

const mapStateToProps = state => {
  return {
    action: state.userRegistration.action
  };
};

const mapDispatchToProps = {
  dispatchLoginAction,
  dispatchRegisterAction
};

export default connect(mapStateToProps, mapDispatchToProps)(LoginTypeCta);
