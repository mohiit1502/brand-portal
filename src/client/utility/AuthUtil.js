import authConfig from "./../config/authorizations";
import Helper from "./helper";

export default class AuthUtil {
  static isActionAccessible (action, userProfile, actedOnRole) {
    if (action && action.indexOf(".") > -1) {
      const actionConfig = Helper.search(action, authConfig);
      if (actionConfig && userProfile && userProfile.role && userProfile.role.name && typeof actionConfig === "object") {
        const role = userProfile.role.name;
        if (actionConfig.length !== undefined) {
          return actionConfig.includes(role);
        } else {
          if (Object.keys(actionConfig).includes(role) && actedOnRole && actionConfig[role].includes(actedOnRole)) {
            return true;
          }
        }
      }
    }
    return false;
  }
}
