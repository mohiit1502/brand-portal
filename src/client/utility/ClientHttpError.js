export default class ClientHttpError {

  status;
  error;
  message;

  constructor (status, error, message) {
    this.status = status;
    this.error = error;
    this.message = message;
  }

}
