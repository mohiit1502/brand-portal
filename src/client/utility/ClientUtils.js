import queryString from "query-string";

export default class ClientUtils {

  static getQueryParams (location) {
    return queryString.parse(location.search);
  }
}
