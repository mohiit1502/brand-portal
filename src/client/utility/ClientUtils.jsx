import React from "react";
import queryString from "query-string";
import {Redirect} from "react-router";

export default class ClientUtils {

  static getQueryParams (location) {
    return queryString.parse(location.search);
  }

  static  getAllValuesFromRecursiveTree(tree) {

    if (typeof tree !== "object" || tree === null) {
      return [tree];
    }

    const list = [];

    for (const i in tree) {
      if (tree.hasOwnProperty(i)) {
        list.push(...this.getAllValuesFromRecursiveTree(tree[i]));
      }
    }
    return list;
  }

  static where (arrayList, filterObj) {
    let index = -1;
    for (let i = 0; i < arrayList.length; i++) {
      for (const j in filterObj) {
        if (filterObj.hasOwnProperty(j) && arrayList[i].hasOwnProperty(j) && filterObj[j] === arrayList[i][j]) {
          index = i;
        } else {
          index = -1;
        }
      }
      if (index !== -1) {
        return index;
      }
    }
    return index;
  }

}
