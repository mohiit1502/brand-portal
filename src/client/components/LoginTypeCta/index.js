import LoginTypeCta from "./LoginTypeCta.jsx";

export default LoginTypeCta;
