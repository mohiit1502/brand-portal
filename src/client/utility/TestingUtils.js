import {applyMiddleware, compose, createStore} from "redux";
import rootReducer from "../reducers";
import thunk from "redux-thunk";
import Immutable from "immutable";
import configureMockStore from 'redux-mock-store';
import ClientHttpError from "./ClientHttpError";

export const findByTestAttribute = (component,attr) => component.find(`[data-test="${attr}"]`);

export const testStore = state => configureMockStore()(state);

export const realStore = state => {
  const configureStore = initialState => {
    const composeEnhancers =
      typeof window === "object" &&
      window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
        window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
          serialize: { // prettier-ignore
            immutable: Immutable
          }
        }) : compose;
    const store = createStore(rootReducer, initialState, composeEnhancers(applyMiddleware(thunk)));

    if (module.hot) {
      module.hot.accept("./reducers", () => {
        const nextRootReducer = require("./reducers").default;
        store.replaceReducer(nextRootReducer);
      });
    }

    return store;
  };

  return configureStore(state || window.__WML_REDUX_INITIAL_STATE__);
}

export const clearKeys = (tree, arr) => {
  if (arr && arr.indexOf(tree) > -1) {
    return;
  }
  if (tree && typeof tree === 'object') {
    if (tree.inputId) {
      return;
    }
    arr.push(tree)
    if (tree.key) {
      try {
        delete tree.key;
      } catch (e) {}
    }
    if (Array.isArray(tree)) {
      tree.forEach(obj => clearKeys(obj, arr));
    } else {
      Object.keys(tree).forEach(key => clearKeys(tree[key], arr));
    }
  }
}

export function setupFetchStub(data) {
  return function fetchStub(_url) {
    return new Promise((resolve) => {
      resolve({
        json: () =>
          Promise.resolve({
            data,
          }),
      })
    })
  }
}

export function setupFetchThrowStub() {
  const error = new ClientHttpError(500, "test-error"); // Directly throwing new ClientHttpError fails in Jest
  throw error;
}
