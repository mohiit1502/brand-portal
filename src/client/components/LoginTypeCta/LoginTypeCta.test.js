/* eslint-disable filenames/match-regex, no-unused-vars, no-undef */
import React, {useRef} from "react";
import LoginTypeCta from "./LoginTypeCta";
import {testStore} from "../../utility/TestingUtils";
import {BrowserRouter} from "react-router-dom";
import {Provider} from "react-redux";
import loginTypeCtaMock from "../../../../test/client/mocks/loginTypeCta";
import {mount} from "enzyme";
import toJson from "enzyme-to-json";

let store;
const setUp = (props,bool) => {
  const mockStore = {
    userRegistration :{
      action: "login"
    }
  };
  const mockStore2 = {
    userRegistration :{
    }
  }
  store = testStore(bool ? mockStore : mockStore2);
  return mount(<Provider store={store}><BrowserRouter><LoginTypeCta {...props} /></BrowserRouter></Provider>);
};

jest.mock("react", () => {
  const originReact = jest.requireActual("react");
  const mUseRef = jest.fn();
  return {
    ...originReact,
    useRef: mUseRef
  };
});

describe("Login Type CTA FAQ Tests", () => {

  let wrapper;
  beforeEach(() => {
    jest.restoreAllMocks();
    jest.resetAllMocks();
  });

  it("renders without error", () => {
    const mRef = {current: document.createElement("div")};
    useRef.mockReturnValue(mRef);
    wrapper = setUp(loginTypeCtaMock["noError"],true);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it("renders unauthorised", () => {
    const mRef = {current: document.createElement("div")};
    useRef.mockReturnValue(mRef);
    wrapper = setUp(loginTypeCtaMock["unauthorised"],false);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
