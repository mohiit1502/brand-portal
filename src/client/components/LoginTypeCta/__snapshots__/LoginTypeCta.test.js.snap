// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Type CTA FAQ Tests renders unauthorised 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(LoginTypeCta)
        action="login"
        history={
          Object {
            "action": "POP",
            "length": 18,
            "location": Object {
              "hash": "",
              "pathname": "/login",
              "search": "?unauthorized",
            },
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/login",
            "search": "?unauthorized",
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/login",
            "url": "/login",
          }
        }
        route={
          Object {
            "exact": true,
            "path": "/login",
          }
        }
      >
        <LoginTypeCta
          dispatchLoginAction={[Function]}
          dispatchRegisterAction={[Function]}
          history={
            Object {
              "action": "POP",
              "length": 18,
              "location": Object {
                "hash": "",
                "pathname": "/login",
                "search": "?unauthorized",
              },
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/login",
              "search": "?unauthorized",
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/login",
              "url": "/login",
            }
          }
          route={
            Object {
              "exact": true,
              "path": "/login",
            }
          }
        >
          <div
            className="c-LoginTypeCta"
          >
            <div
              className="page-container css-0 d-flex flex-column"
            >
              <div
                className="logoBox"
              >
                <img
                  alt="WBP Dark Logo"
                  className="logo"
                  src="BP_Logo_darkBG.svg"
                />
              </div>
              <div
                className="main-container position-absolute"
              >
                <div
                  className="content-container loader"
                >
                  <div
                    className="padder p-4"
                  >
                    <div
                      className="tab-content text-center"
                    >
                      <span
                        className="page-header mb-2"
                      >
                        <span />
                      </span>
                      <p
                        className="sub-title mb-0"
                      >
                        Streamline 
                        registration
                         by using one of your existing
                      </p>
                      <p
                        className="sub-title"
                      >
                        Walmart accounts.
                      </p>
                      <a
                        className="app-btn d-block mt-2"
                        href="/api/falcon/login?clientType=seller"
                        onClick={[Function]}
                      />
                      <div
                        className="OrSeparator w-100 d-flex my-4"
                      >
                        <hr />
                        <span>
                          OR
                        </span>
                        <hr />
                      </div>
                      <a
                        className="app-btn d-block secondary-btn mt-2"
                        href="/api/falcon/register?clientType=supplier"
                        onClick={[Function]}
                      />
                    </div>
                  </div>
                </div>
                <div>
                  <div
                    className="footnote-container mt-4"
                  >
                    <span
                      className="footnote d-block text-center"
                    >
                      We recommend setting your monitor resolution to at least 1024 x 768
                    </span>
                    <span
                      className="footnote d-block text-center"
                    >
                      U.S. Users: Please use Chrome v49 or newer
                    </span>
                    <span
                      className="footnote d-block text-center"
                    >
                      International Users: Please use Internet Explorer
                    </span>
                    <a
                      className="footnote d-block text-center policy-link"
                      href="https://corporate.walmart.com/privacy-security"
                      target="_blank"
                    >
                      Privacy Policy
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </LoginTypeCta>
      </Connect(LoginTypeCta)>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Login Type CTA FAQ Tests renders without error 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(LoginTypeCta)
        history={
          Object {
            "action": "PUSH",
            "length": 17,
            "location": Object {
              "hash": "",
              "key": "3r32hc",
              "pathname": "/login",
              "search": "",
            },
          }
        }
        location={
          Object {
            "hash": "",
            "key": "3r32hc",
            "pathname": "/login",
            "search": "",
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/login",
            "url": "/login",
          }
        }
        route={
          Object {
            "exact": true,
            "path": "/login",
          }
        }
      >
        <LoginTypeCta
          action="login"
          dispatchLoginAction={[Function]}
          dispatchRegisterAction={[Function]}
          history={
            Object {
              "action": "PUSH",
              "length": 17,
              "location": Object {
                "hash": "",
                "key": "3r32hc",
                "pathname": "/login",
                "search": "",
              },
            }
          }
          location={
            Object {
              "hash": "",
              "key": "3r32hc",
              "pathname": "/login",
              "search": "",
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/login",
              "url": "/login",
            }
          }
          route={
            Object {
              "exact": true,
              "path": "/login",
            }
          }
        >
          <div
            className="c-LoginTypeCta"
          >
            <div
              className="page-container css-0 d-flex flex-column"
            >
              <div
                className="logoBox"
              >
                <img
                  alt="WBP Dark Logo"
                  className="logo"
                  src="BP_Logo_darkBG.svg"
                />
              </div>
              <div
                className="main-container position-absolute"
              >
                <div
                  className="content-container loader"
                >
                  <div
                    className="padder p-4"
                  >
                    <div
                      className="tab-content text-center"
                    >
                      <span
                        className="page-header mb-2"
                      >
                        <span>
                          Log In
                        </span>
                      </span>
                      <p
                        className="sub-title mb-0"
                      >
                        Streamline 
                        login
                         by using one of your existing
                      </p>
                      <p
                        className="sub-title"
                      >
                        Walmart accounts.
                      </p>
                      <a
                        className="app-btn d-block mt-2"
                        href="/api/falcon/login?clientType=seller"
                        onClick={[Function]}
                      >
                        Use Seller Center Account
                      </a>
                      <div
                        className="OrSeparator w-100 d-flex my-4"
                      >
                        <hr />
                        <span>
                          OR
                        </span>
                        <hr />
                      </div>
                      <a
                        className="app-btn d-block secondary-btn mt-2"
                        href="/api/falcon/login?clientType=supplier"
                        onClick={[Function]}
                      >
                        Continue with Brand Portal Log In
                      </a>
                    </div>
                  </div>
                </div>
                <div>
                  <div
                    className="footnote-container mt-4"
                  >
                    <span
                      className="footnote d-block text-center"
                    >
                      We recommend setting your monitor resolution to at least 1024 x 768
                    </span>
                    <span
                      className="footnote d-block text-center"
                    >
                      U.S. Users: Please use Chrome v49 or newer
                    </span>
                    <span
                      className="footnote d-block text-center"
                    >
                      International Users: Please use Internet Explorer
                    </span>
                    <a
                      className="footnote d-block text-center policy-link"
                      href="https://corporate.walmart.com/privacy-security"
                      target="_blank"
                    >
                      Privacy Policy
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </LoginTypeCta>
      </Connect(LoginTypeCta)>
    </Router>
  </BrowserRouter>
</Provider>
`;
